source/Cosmos.System2/Graphics/SVGAIICanvas.cs

```diff
@@ -92,21 +106,28 @@ public override void DrawPoint(Pen aPen, int aX, int aY)
         {
             Color xColor = aPen.Color;
 
-            if (xColor.A == 0)
+            if (DoubleBuffered)
             {
-                return;
+                _xSVGADriver.SetPixel((uint)aX, (uint)aY, (uint)xColor.ToArgb());
             }
-            else if (xColor.A < 255)
+            else
             {
-                xColor = AlphaBlend(xColor, GetPointColor(aX, aY), xColor.A);
+                if (xColor.A == 0)
+                {
+                    return;
+                }
+                else if (xColor.A < 255)
+                {
+                    xColor = AlphaBlend(xColor, GetPointColor(aX, aY), xColor.A);
+                }
+
+                mSVGAIIDebugger.SendInternal($"Drawing point to x:{aX}, y:{aY} with {xColor.Name} Color");
+                _xSVGADriver.SetPixel((uint)aX, (uint)aY, (uint)xColor.ToArgb());
+                mSVGAIIDebugger.SendInternal($"Done drawing point");
+                /* No need to refresh all the screen to make the point appear on Screen! */
+                //xSVGAIIDriver.Update((uint)x, (uint)y, (uint)mode.Columns, (uint)mode.Rows);
+                _xSVGADriver.Update((uint)aX, (uint)aY, 1, 1);
```

Comment:
I think it would be better if this is the only line which is not executed when DoubleBuffered, is there a reason to not have AlphaBlending when double buffering?

Reply:
it can be used for double buffering too but it will cause performance issues if we do more if / else / else if ...

Reply:
I just want to move the if in this method to only around the update command and have the drawing have the same logic regardless of double buffering.